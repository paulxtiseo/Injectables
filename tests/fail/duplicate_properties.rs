use injectables::{injectable, inject_fields};

#[injectable]
pub struct Base {
  pub id: u64,
}

#[inject_fields(Base)]
pub struct Conflict {
  pub id: u32,  // Should fail because 'id' is already injected
}

fn main() {}