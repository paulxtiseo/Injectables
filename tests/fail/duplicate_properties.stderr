error[E0124]: field `id` is already declared
  --> tests/fail/duplicate_properties.rs:8:1
   |
8  | #[inject_fields(Base)]
   | ^^^^^^^^^^^^^^^^^^^^^^ field already declared
9  | pub struct Conflict {
10 |   pub id: u32,  // Should fail because 'id' is already injected
   |   ----------- `id` first declared here
   |
   = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
